src/twisted/test/test_application.py

```diff
@@ -889,3 +891,54 @@ def install():
                                ['--reactor', 'fakereactortest', 'subcommand'])
         self.assertIn(message, e.args[0])
         self.assertIn("help-reactors", e.args[0])
+
+
+
+class HelpReactorsTests(unittest.TestCase):
+    """
+    --help-reactors lists the available reactors
+    """
+    def setUp(self):
+        """
+        Get the text from --help-reactors
+        """
+        self.options = app.ReactorSelectionMixin()
+        self.options.messageOutput = NativeStringIO()
+        self.assertRaises(SystemExit, self.options.opt_help_reactors)
+        self.message = self.options.messageOutput.getvalue()
+
+
+    def test_asyncIOPy2(self):
+        """
+        --help-reactors should NOT display the asyncIO reactor on Python 2
+        """
+        self.assertNotIn(twisted_reactors.asyncio.description, self.message)
+    if _PY3:
+        test_asyncIOPy2.skip = "Not applicable on Python 3"
```

Comment:
I might instead write this and the following test as a single test with the `if _PY3: ...` conditional inside the test method, switching between either asserting in or not in depending on.  This is an optional change.